Add regex_dna benchmark

haypo opened the pull request:
Patch written by Serhiy Storchaka: http://bugs.python.org/issue26436

It seems like Pyston has the same benchmark with minor changes:
* https://github.com/dropbox/pyston-perf/blob/master/benchmarking/benchmark_suite/fasta.py
* Pyston benchmark is called "fasta"
* fasta replaces stdout to check the output
* fasta moved the LCG RNG to a genRandom() subfunction which is a good idea to make random_fasta() more readable. Since it's only used to initialize the benchmark, it has no impact on performance.
* probably other subtle differences, I didn't check carefully

serhiy-storchaka commented:
"fasta" is used just for generating input data for regex-dna. The implementation in my patch is hardly optimized, since the initialization time is not such small.

serhiy-storchaka commented on performance/benchmarks/bm_regex_dna.py:
```diff
@@ -0,0 +1,213 @@
+
+    # pRNG Vars
+    im = 139968.0
+    #seed = 42.0
```
This can be removed.

haypo replied:
The variable is still used. Do you suggest to replace the variable by its value in the code?

haypo commented:
> The implementation in my patch is hardly optimized, since the initialization time is not such small.

Oh right, it takes ~300 ms on CPython 3.5 on my laptop. By the way, I "optimized" the benchmark to only call init_benchmark() once. Before, it was initialized once per sample: perf uses 4 samples per process by default (including 1 warmup sample).

haypo commented:
I merged the pull request. We still can push minor enhancements later, like the comment on init_benchmarks(). Thanks Serhiy!

One less bechmark to port from Pyston (in Pyston, the benchmark is called "fasta").

serhiy-storchaka commented:
> One less bechmark to port from Pyston (in Pyston, the benchmark is called
> "fasta").

The "fasta" benchmark and the "regex-dna" benchmark are *different* benchmarks.

serhiy-storchaka commented:
> +    #seed = 42.0
> 
> The variable is still used. Do you suggest to replace the variable by its
> value in the code?

I meant only the commented out assignment.

haypo commented:
> The "fasta" benchmark and the "regex-dna" benchmark are *different* benchmarks.

Oops, I fixed the TODO document.

> I meant only the commented out assignment.

Ah. I replaced the constant with a command line option which has a default value of 42 :-)

haypo commented:
I consider that the issue (pull request) is done. If you have more remarks, you might open a new issue or send a new pull request.
